### PR TITLE
new branch for Typescript: Mentors Dashboard

### LINKED ISSUE #1615: TypeScript: Mentors Dashboard
image

### PR DESCRIPTION
**Pull request by d-leeee**

### CODE REVIEW

File: src/data/admin/Mentors.tsx [hunk 1]
```diff
@@ -14,39 +36,36 @@ export const TAGS = [
   },
 ];
 
-export const COLUMNS = [
+export const COLUMNS: extendedColumnDef<Person, string>[] = [
   generateSelect(),
   {
     accessorKey: "name",
     header: "Name",
     width: "w-3/12",
```
Comment (d-leeee) [inline src/data/admin/Mentors.tsx]: can also use meta: { width: "" } and then change to column.columnDef.meta?.width in Table.jsx
this way we don't need to create another type that extends ColumnDef

File: src/data/admin/Mentors.tsx [hunk 2]
```diff
@@ -2,8 +2,30 @@ import Checkbox from "@/components/Checkbox";
...
+  text: string;
+  value: number;
+};
+
+type objectProp = {
+  availability: string[];
+  response: string;
+};
+
+type extendedColumnDef<TData, TValue> = ColumnDef<TData, TValue> & {
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx]: use an interface and extends

File: src/data/admin/Mentors.tsx (hunk 1 above)
Comment (shahdivyank) [reply src/data/admin/Mentors.tsx]: go for it, this change will also have to be propagated to the actual dashboards

File: src/data/admin/Mentors.tsx [hunk 3]
```diff
@@ -14,39 +36,36 @@ export const TAGS = [
...
   },
   {
     accessorKey: "email",
     header: "Email",
     width: "w-4/12",
     enableColumnFilter: true,
     filterFn: "includesString",
-    searchable: true,
-    cell: ({ getValue }) => <div>{getValue()}</div>,
+    cell: (props: { getValue: () => string }) => <div>{props.getValue()}</div>,
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx]: extract the types out

File: src/data/admin/Mentors.tsx [hunk 4]
```diff
@@ -56,6 +75,7 @@ export const DROPDOWN = ({ object }) => {
               toggle={object.availability.includes(text)}
               text={text}
               key={index}
+              color={"bg-hackathon-blue-100"}
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx]: shouldn't be needed?

File: src/data/admin/Mentors.tsx [hunk 5]
```diff
@@ -2,8 +2,30 @@ import Checkbox from "@/components/Checkbox";
 import { AVAILABILITY } from "../form/Information";
 import { generateSelect, generateStatus } from "./Columns";
 import { STATUSES } from "@/data/Statuses";
+import { ColumnDef } from "@tanstack/react-table";
 
-export const TAGS = [
+type Person = {
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx]: create a mentor type, check the attributes that we accept for the mentors

File: src/data/admin/Mentors.tsx (hunk 4 above)
Comment (d-leeee) [reply src/data/admin/Mentors.tsx]: if i remove it it says property color is missing but required in type

Comment (shahdivyank) [reply src/data/admin/Mentors.tsx]: it should be made optional then as a type

File: src/data/admin/Mentors.tsx [hunk 6]
```diff
@@ -14,39 +37,36 @@ export const TAGS = [
...
 ];
 
-export const COLUMNS = [
+export const COLUMNS: ColumnDef<Mentor, string>[] = [
   generateSelect(),
   {
     accessorKey: "name",
     header: "Name",
-    width: "w-3/12",
+    meta: { width: "w-3/12" },
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx:43]: this change would need to be propagated throughout all the dashboards

**Comment by shahdivyank** [comment]
close the old pr if it isn't being used

File: src/data/admin/Mentors.tsx (hunk 4 above)
Comment (d-leeee) [reply src/data/admin/Mentors.tsx]: should i do this inside Checkbox.jsx or Mentors.tsx

File: src/data/admin/Mentors.tsx [hunk 7]
```diff
@@ -2,8 +2,31 @@ import Checkbox from "@/components/Checkbox";
...
-export const TAGS = [
+type Mentor = {
+  name: string;
+  email: string;
+  discord: string;
+  availability: string[];
+  response: string;
+};
+
+type cellProp = {
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx]: should come from tanstack

File: src/data/admin/Mentors.tsx [hunk 8]
```diff
@@ -14,39 +37,36 @@ export const TAGS = [
...
     filterFn: "includesString",
-    searchable: true,
-    cell: ({ getValue }) => <div>{getValue()}</div>,
+    cell: (props: cellProp) => <div>{props.getValue()}</div>,
   },
   generateStatus(STATUSES),
 ];
 
-export const DROPDOWN = ({ object }) => {
+export const DROPDOWN = ({ object }: { object: objectProp }) => {
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx]: should be extracted entirely

File: src/data/admin/Mentors.tsx [hunk 9]
```diff
@@ -2,8 +2,31 @@ import Checkbox from "@/components/Checkbox";
...
+  availability: string[];
+  response: string;
+};
+
+type tags = {
+  text: string;
+  value: number;
+};
+
+type objectProp = {
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx]: this should be integrated directly with the below type

**Comment by d-leeee** [comment]
mb with all the mistakes, first time doing react and stuff 😭

**Review by shahdivyank — APPROVED** [review]

**Pull request merged**